shell/platform/glfw/flutter_glfw.cc

```diff
@@ -507,18 +530,24 @@ FlutterDesktopWindowControllerRef FlutterDesktopCreateWindow(
   }
   GLFWClearCanvas(window);
 
+  // Create a state object attached to the window.
+  auto* state = new FlutterDesktopWindowControllerState();
```

Comment:
Please create a unique pointer to the state here. When you are about to transfer ownership of the same out of the method, call release on the pointer. That way, all early returns don't have to remember to keep track of the deletion and the point of ownership transfer is clearly marked.